libraries/DataFlash/DataFlash_File.cpp

```diff
@@ -830,11 +838,18 @@ uint16_t DataFlash_File::get_num_logs()
  */
 void DataFlash_File::stop_logging(void)
 {
+    // best-case effort to avoid annoying the IO thread
+    const bool have_sem = write_fd_semaphore->take(1);
```

Comment:
This will prevent some stuff, but not everything. Are we really sure that's how we want to proceed? I understand it can be problematic because if stop logging was asked for then we should indeed stop it, but I would say that given we have two threads accessing the file, we should try to close it immediately but if we can't (because we can't acquire the semaphore lock) we set a flag to close it as soon as possible.